setup.py

```diff
@@ -50,65 +49,72 @@ def cmake(self):
         cmake_cmd = shutil.which("cmake")
         return cmake_cmd
 
+    @property
+    def make(self):
+        """Find and use installed cmake"""
+        make_cmd = shutil.which("make")
+        return make_cmd
+
     def run(self):
+        # Validate dependencies
+        check_prereq("cmake")
         check_prereq("make")
         check_prereq("gcc")
         check_prereq("g++")
 
+        # Set up parameters
+        source_directory = Path(os.path.abspath(os.path.dirname(__file__))).resolve()
         build_directory = Path(self.build_temp).resolve()
-        cmake_args = [
-            '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=' + str(build_directory),
-            '-DPYTHON_EXECUTABLE=' + sys.executable,
-            '-DSR_PYTHON=ON'
-        ]
-
         cfg = 'Debug' if self.debug else 'Release'
-        build_args = ['--config', cfg]
-        build_args += ['--', f'-j{str(NPROC)}']
-        self.build_args = build_args
-
-        cmake_args += ['-DCMAKE_BUILD_TYPE=' + cfg]
 
-        # setup build environment
+        # Setup build environment
         env = os.environ.copy()
         env['CXXFLAGS'] = '{} -DVERSION_INFO=\\"{}\\"'.format(
             env.get('CXXFLAGS', ''),
             self.distribution.get_version())
 
-        # make tmp dir
-        if not build_directory.is_dir():
-            os.makedirs(self.build_temp)
-
-        # make install dir
-        setup_path = Path(os.path.abspath(os.path.dirname(__file__))).resolve()
-        install_directory = setup_path.joinpath("install")
-        if not install_directory.is_dir():
-            os.makedirs(install_directory)
-
-        print('-'*10, 'Building C dependencies', '-'*40)
-        make_cmd = shutil.which("make")
-
-        # build dependencies
-        subprocess.check_call([f"{make_cmd} deps"],
-                              cwd=setup_path,
-                              shell=True)
-
-        # run cmake prep step
-        print('-'*10, 'Running CMake prepare', '-'*40)
-        subprocess.check_call([self.cmake, setup_path] + cmake_args,
-                              cwd=build_directory,
-                              env=env)
-
-
-        print('-'*10, 'Building extensions', '-'*40)
-        cmake_cmd = [self.cmake, '--build', '.'] + self.build_args
-        subprocess.check_call(cmake_cmd,
-                              cwd=build_directory)
-
-        shutil.copytree(setup_path.joinpath("install"),
-                        build_directory.joinpath("install"))
+        # Build dependencies
+        print('-'*10, 'Building third-party dependencies', '-'*40)
+        subprocess.check_call(
+            [f"{self.make}", "deps"],
```

Comment:
Super minor thing here:
It looks like `self.make` should already be returning a `str | None` and we know that it will not be `None` bc the previous `check_prereq` did not error out, so the `f"{self.make}"` looks like a redundant cast to a string.